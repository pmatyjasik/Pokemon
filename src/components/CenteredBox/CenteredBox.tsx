"use client";

import { Box, useTheme } from "@mui/material";
import { ReactNode } from "react";

interface CenteredBoxProps {
    children: ReactNode;
}

export const CenteredBox = ({ children }: CenteredBoxProps) => {
    const { palette, shape, spacing } = useTheme();

    return (
        <Box
            display='flex'
            justifyContent='center'
            alignItems='center'
            minHeight='100vh'
        >
            <Box
                component='div'
                border={1}
                borderColor={palette.grey[300]}
                borderRadius={shape.borderRadius}
                maxWidth={spacing(68)}
                width={spacing(68)}
                padding={spacing(4)}
            >
                {children}
            </Box>
        </Box>
    );
};
