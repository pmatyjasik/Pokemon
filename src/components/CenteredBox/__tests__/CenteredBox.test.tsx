import { cleanup, render, screen } from "@testing-library/react";
import { ReactNode } from "react";
import { CenteredBox } from "@/components/CenteredBox/CenteredBox";

const renderComponent = (children: ReactNode) => {
    return render(<CenteredBox>{children}</CenteredBox>);
};

const testValue = "Test Content";

describe("CenteredBox Component", () => {
    afterEach(() => {
        cleanup();
    });

    it("Matches DOM Snapshot", () => {
        const { asFragment } = renderComponent(<div>{testValue}</div>);

        expect(asFragment()).toMatchSnapshot();
    });

    it("renders children correctly", () => {
        renderComponent(<div data-testid='child'>{testValue}</div>);

        expect(screen.getByTestId("child")).toBeInTheDocument();
        expect(screen.getByText(testValue)).toBeInTheDocument();
    });

    it("centers the content vertically and horizontally", () => {
        renderComponent(<div>{testValue}</div>);

        const outerBox = screen.getByText(testValue).parentElement?.parentElement;

        expect(outerBox).toHaveStyle("display: flex");
        expect(outerBox).toHaveStyle("justify-content: center");
        expect(outerBox).toHaveStyle("align-items: center");
        expect(outerBox).toHaveStyle("min-height: 100vh");
    });
});
