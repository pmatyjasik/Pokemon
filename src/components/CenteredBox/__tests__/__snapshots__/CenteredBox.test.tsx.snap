// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenteredBox Component Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-ue6sde"
  >
    <div
      class="MuiBox-root css-7ksecd"
    >
      Test Content
    </div>
  </div>
</DocumentFragment>
`;
